syft/pkg/license.go

```diff
@@ -0,0 +1,157 @@
+package pkg
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/license"
+	"github.com/anchore/syft/syft/source"
+)
+
+var _ sort.Interface = (*Licenses)(nil)
+
+// License represents an SPDX Expression or license value extracted from a packages metadata
+// We want to ignore URL and Location since we will always merge these fields
+// a License is a unique combination of value, expression and type
+// its sources are always considered merge and additions to the evidence
+// of where it was found and how it was sourced.
+// This is different from how we treat a package since we consider package paths to distinguish
+// if packages should be kept separate, this is different for licenses since we're only looking for evidence
+// of where a license was declared/concluded for a package
+type License struct {
+	Value          string             `json:"value"`
+	SPDXExpression string             `json:"spdxExpression"`
+	Type           license.Type       `json:"type"`
+	URL            internal.StringSet `hash:"ignore"`
```

Comment:
I think this might be covered by the main `License` comment above

"We want to ignore URL and Location...", but I can update the wording there if it's unclear!

Took another stab at the wording with an updated commit